cmd/connection.go

```diff
@@ -0,0 +1,469 @@
+		Example: `
+	privx-cli connections access-roles [access flags] --conn-id <CONN-ID>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return accessRoleList(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.connID, "conn-id", "", "connection ID")
+	cmd.MarkFlagRequired("conn-id")
+
+	return cmd
+}
+
+func accessRoleList(options connectionOptions) error {
+	api := connectionmanager.New(curl())
+
+	roles, err := api.AccessRoles(options.connID)
+	if err != nil {
+		return err
+	}
+
+	return stdout(roles)
+}
+
+//
+//
+func connectionAccessRoleGrantCmd() *cobra.Command {
+	options := connectionOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "grant-access-role",
+		Short: "Add the role to a list of roles that can explicitly access this connection data",
+		Long:  `Add the role to a list of roles that can explicitly access this connection data`,
+		Example: `
+	privx-cli connections grant-role [access flags] --conn-id <CONN-ID> --role-id <ROLE-ID>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return connectionAccessRoleGrant(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.connID, "conn-id", "", "connection ID")
+	flags.StringVar(&options.roleID, "role-id", "", "role ID")
+	cmd.MarkFlagRequired("conn-id")
+	cmd.MarkFlagRequired("role-id")
+
+	return cmd
+}
+
+func connectionAccessRoleGrant(options connectionOptions) error {
+	api := connectionmanager.New(curl())
+
+	err := api.GrantAccessRoleToConnection(options.connID, options.roleID)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+//
+//
+func connectionAccessRoleRevokeCmd() *cobra.Command {
+	options := connectionOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "revoke-access-role",
+		Short: "Remove the role from a list of roles that can explicitly access this connection data",
+		Long:  `Remove the role from a list of roles that can explicitly access this connection data`,
+		Example: `
+	privx-cli connections revoke-access-role [access flags] --conn-id <CONN-ID> --role-id <ROLE-ID>
+		`,
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return connectionAccessRoleRevoke(options)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&options.connID, "conn-id", "", "connection ID")
+	flags.StringVar(&options.roleID, "role-id", "", "role ID")
+	flags.BoolVarP(&options.force, "force", "f", false, "force command")
+	cmd.MarkFlagRequired("role-id")
+
+	return cmd
+}
+
+func connectionAccessRoleRevoke(options connectionOptions) error {
+	api := connectionmanager.New(curl())
+
+	if options.connID != "" {
+		err := api.RevokeAccessRoleFromConnection(options.connID, options.roleID)
+		if err != nil {
+			return err
+		}
+	}
+
+	if options.connID == "" {
+		if !options.force {
+			fmt.Println("You are about to delete the roles from ALL the connections. Please use the --force | -f flag to proceed with the command")
```

Comment:
Print to stderr. Message could be improved to something similar to this: "Error: this action will revoke data access right from this role to all connections.\nUse --force | -f flag to revoke data access to all connections or use --conn-id to revoke data access to a specific connection"...